Entities.js

```diff
@@ -0,0 +1,339 @@
+import { StyleSheet, Text, View, Image } from 'react-native';
+import { Dimensions, Keyboard } from 'react-native';
+import { React, Component, useEffect, useState, useRef } from 'react';
+import { UseOnKeyPress, UseOnKeyRelease } from "./Key.js"
+import { CreateCircle, CreateBrick, DeleteEntity } from "./Dodge.js"
+import Matter from "matter-js"
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+let counter = 0
+let GlobalX = 400
+let GlobalY = 400
+let playerHeight = 150
+let playerWidth = 150
+let input = [0,0,0,0]
+//      left,right,up,down (w,a,s,d) each stores a value, 0 = not pressed
+
+//count intersections with players
+const Counter = () => {
+  console.log({counter})
+} 
+
+//function to detect wasd input on computer
+function MoveKnight (){
+  const Left = () => {
+    input[0] = 1.5
+  }
+  const Right = () => {
+    input[1] = 1.5  
+  }
+  const Up = () => {
+    input[2] = 1.5
+  }
+  const Down = () => {
+    input[3] = 1.5
+  }
+  const UndoLeft = () => {
+    input[0] = 0
+  }
+  const UndoRight = () => {
+    input[1] = 0  
+  }
+  const UndoUp = () => {
+    input[2] = 0
+  }
+  const UndoDown = () => {
+    input[3] = 0
+  }
+  UseOnKeyPress(Left, 'a')
+  UseOnKeyPress(Right,'d')
+  UseOnKeyPress(Up, 'w')
+  UseOnKeyPress(Down, 's')
+  UseOnKeyRelease(UndoLeft,'a')
+  UseOnKeyRelease(UndoRight,'d')
+  UseOnKeyRelease(UndoUp, 'w')
+  UseOnKeyRelease(UndoDown, 's')
+}
+
+//main player character
+class Knight{
+
+  constructor(){
+    this.height = playerHeight;
+    this.width = playerWidth;
+    this.backgroundColor = "blue";
+    this.x = 400;
+    this.y = 400;
+    this.velocityX = 0;
+    this.velocityY = 0;
+    this.Image = "./assets/Background.jpeg";
+    this.renderer = <CreateBrick/>;
+    this.momentum = [0,0];
```

Comment:
You could make this an object with names, i.e.:
```
this.momentum = {x: 0, y: 0};
```
Elsewhere:
```
this.momentum.x += 3;
```

Reply:
I think I'm fine with leaving it as an array, to me it is pretty clear that it is x and then y